### PR TITLE
TOOL-974 export artifact zip for the ship

### PR DESCRIPTION
**Pull request by BirmacherAkos**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -283,6 +282,8 @@ func main() {
 	archiveZipPath := filepath.Join(cfg.OutputDir, cfg.ArtifactName+".xcarchive.zip")
 	ideDistributionLogsZipPath := filepath.Join(cfg.OutputDir, "xcodebuild.xcdistributionlogs.zip")
 
+	shipZipPath := filepath.Join(os.Getenv("BITRISE_DEPLOY_DIR"), cfg.ArtifactName+".xcarchive.zip")
```
Comment (trapacska) [inline main.go]: "ship" has no meaning in this context, this should be archiveExportPath or similar

File: main.go [hunk 2]
```diff
@@ -283,6 +282,8 @@ func main() {
 	archiveZipPath := filepath.Join(cfg.OutputDir, cfg.ArtifactName+".xcarchive.zip")
 	ideDistributionLogsZipPath := filepath.Join(cfg.OutputDir, "xcodebuild.xcdistributionlogs.zip")
 
+	xcarchiveZipPath := filepath.Join(os.Getenv("BITRISE_DEPLOY_DIR"), cfg.ArtifactName+".xcarchive.zip")
```
Comment (lpusok) [inline main.go]: Can you ensure that the zip name is unique by generating a custom artifact name? Also it seems cfg.ArtifactName is not validated to be not empty.

File: main.go [hunk 3]
```diff
@@ -72,10 +72,9 @@ type configs struct {
 	XcodebuildOptions         string `env:"xcodebuild_options"`
 	DisableIndexWhileBuilding bool   `env:"disable_index_while_building,opt[yes,no]"`
 
-	IsExportXcarchiveZip string `env:"is_export_xcarchive_zip,opt[yes,no]"`
```
Comment (lpusok) [inline main.go]: Should not we print a warning that the input is deprecated?

File: main.go [hunk 4]
```diff
@@ -869,12 +870,14 @@ is available in the $BITRISE_IDEDISTRIBUTION_LOGS_PATH environment variable`)
...
-			fail("Failed to export %s, error: %s", bitriseXCArchiveZipPthEnvKey, err)
-		}
+	if err := utils.ExportOutputDirAsZip(tmpArchivePath, archiveZipPath, bitriseXCArchiveZipPthEnvKey); err != nil {
+		fail("Failed to export %s, error: %s", bitriseXCArchiveZipPthEnvKey, err)
+	}
+
+	log.Donef("The xcarchive zip path is now available in the Environment Variable: %s (value: %s)", bitriseXCArchiveZipPthEnvKey, archiveZipPath)
 
-		log.Donef("The xcarchive zip path is now available in the Environment Variable: %s (value: %s)", bitriseXCArchiveZipPthEnvKey, archiveZipPath)
+	if err := command.CopyFile(archiveZipPath, xcarchiveZipPath); err != nil {
```
Comment (lpusok) [inline main.go]: In case of the default input value for output dir, this will copy from the deploy dir to the deploy dir.

File: main.go (hunk 2 above)
Comment (lpusok) [reply main.go]: Discussed the unique name and agreed that is not the best user experience. Still think that should handle the case if the artifact name is empty.

File: step.yml [hunk 5]
```diff
@@ -266,16 +266,6 @@ inputs:
       - xcpretty
       - xcodebuild
       is_required: true
-  - is_export_xcarchive_zip: "no"
-    opts:
-      category: Debug
-      title: Export the generated archive?
-      description: |-
-        If this input is set to `yes`, the generated .xcarchive will be zipped and moved to `output_dir`.
```
Comment (lpusok) [inline step.yml]: Please remove line 337 " This environment will be exported only if, is_export_xcarchive_zip="yes""

**Pull request merged**